lte/gateway/c/session_manager/CMakeLists.txt

```diff
@@ -42,7 +42,7 @@ add_definitions(-DLOG_WITH_GLOG)
 
 message("Build type is ${CMAKE_BUILD_TYPE}")
 if (CMAKE_BUILD_TYPE STREQUAL "Debug")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror -Wno-unused-function")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-unused-function")
```

Comment:
Hey @electronjoe, just curious, but per your comment on one of the linked PRs, it was mentioned you observed issues with mconfig while building `test_session_manager` on your docker container. 

Are you observing the same on our agw Vagrant vm when you do:
```
vagrant@magma-dev:~/magma/lte/gateway$ make test_session_manager
```

Reply:
Hi @uri200! Thanks for commenting.

> Hey @electronjoe, just curious, but per your comment on one of the linked PRs, it was mentioned you observed issues with mconfig while building `test_session_manager` on your docker container.
> 
> Are you observing the same on our agw Vagrant vm when you do:
> 
> ```
> vagrant@magma-dev:~/magma/lte/gateway$ make test_session_manager
> ```

Yes, In attempting to migrate to Ubuntu 20.04 I'm currently blocked by -Wdeprecated warnings on **protoc compiled protobuf outputs** that are converted to `-Werror`.  This does not occur int he vagrant VM - which is not Ubuntu 20.04 (which we are migrating to). There is a Ubuntu 20.04 Vagrant floating around, I have not tracked it down yet to determine if and how this was addressed.

> If not, maybe the issue is different gcc versions?

This is gcc 9.3 (the baseline for Ubuntu 20.04) - but the issue is with the -Werror flag being applied to protobuf compiles (which is not the case in our older Ubunto 18.04 Vagrant VMs).  Ultimately this points to a problem with our CMake configurations and / or a problem with Cmake - after moving forward with the CMake bundled with Ubuntu 20.04.  Looking at how we compile these protos - it's not surprising to me that this is a tangle (compiler flags leaking / inconsistently applied across build targets and build environments).

Reply:
@uri200 you prompted me to get GitHub GCC Warning annotations working for Pull Request code reviews.  I think this clears the way quite well for this PR to land, removing `-Werror`?

https://github.com/magma/magma/pull/5404

Reply:
Sound like a good solution to unblock the build and still keep the warnings visibility👍